### PR TITLE
Update to v8.4.0

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
We seem to be having some sort of a `libstdc++` vs. `libc++` issue under OSX (see https://github.com/libspatialindex/libspatialindex/issues/123 for similar errors).  Not sure what to make of it, though.

**Comment by ocefpaf** [comment]
> We seem to be having some sort of a `libstdc++` vs. `libc++` issue under OSX (see [libspatialindex/libspatialindex#123](https://github.com/libspatialindex/libspatialindex/issues/123) for similar errors). Not sure what to make of it, though.

Wow. That is something that happened at the earlier days of conda-forge and we built tons of tools to avoid it.

**Comment by xylar** [comment]
Errors are:
```
Undefined symbols for architecture x86_64:
  "__ZNKSt3__16locale9use_facetERNS0_2idE", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNKSt3__18ios_base6getlocEv", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEE3putEc", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
  "__ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEE5flushEv", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
  "__ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEE6sentryC1ERS3_", referenced from:
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEE6sentryD1Ev", referenced from:
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEElsEi", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
  "__ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEElsEm", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
  "__ZNSt3__14coutE", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
  "__ZNSt3__15ctypeIcE2idE", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNSt3__16localeD1Ev", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNSt3__18ios_base33__set_badbit_and_consider_rethrowEv", referenced from:
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZNSt3__18ios_base5clearEj", referenced from:
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "__ZSt9terminatev", referenced from:
      ___clang_call_terminate in ESMCI_StringSubr.o
  "__ZdlPv", referenced from:
      __ZNSt3__116__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_ in ESMCI_StringSubr.o
  "__Znwm", referenced from:
      __ZNSt3__116__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_ in ESMCI_StringSubr.o
  "___cxa_begin_catch", referenced from:
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
      ___clang_call_terminate in ESMCI_StringSubr.o
  "___cxa_end_catch", referenced from:
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
  "___gxx_personality_v0", referenced from:
      _c_strings_ in ESMCI_StringSubr.o
      __ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m in ESMCI_StringSubr.o
      __ZNSt3__116__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_ in ESMCI_StringSubr.o
      Dwarf Exception Unwind Info (__eh_frame) in ESMCI_StringSubr.o
```

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
I think I understand the problem.  We need to explicitly link `-lc++` when we build fortran code.  It's included automatically with c++ code but not with fortran, at least not on OSX with MPI.

**Comment by xylar** [comment]
OMG, OMG, it's working!

**Comment by xylar** [comment]
Rebased to handle merge conflict.  I'm going to add the auto-merge tag.

File: recipe/build.sh [hunk 1]
```diff
@@ -48,7 +49,7 @@ fi
 
 if [[ $(uname) == Darwin ]]; then
   export LDFLAGS="-headerpad_max_install_names $LDFLAGS"
-  export ESMF_F90LINKOPTS="$LDFLAGS -pthread"
+  export ESMF_F90LINKOPTS="$LDFLAGS -pthread -lc++"
```
Comment (xylar) [inline recipe/build.sh:52]: @ocefpaf and @rokuingh, let me know if you're concerned about this fix.

**Pull request merged**